### DISCUSSION TITLE
How to get workspace all symbol, in init request after.

### DISCUSSION THREAD

**Discussion by Yurii-huang** [clangd/clangd#1175] [Q&A] [answered]
After initialization, no files were opened. Allow all symbols to be obtained from "Clangd"?

I see links: [symbols.test](https://github.com/llvm/llvm-project/blob/d480f968ad8b56d3ee4a6b6df5532d485b0ad01e/clang-tools-extra/clangd/test/symbols.test). "didopen" is required before "workspace/symbol" can be used normally. the same of my local test result.

I see that many IDEs have the function of symbol retrieval for all files under the project. I'm not sure whether this is supported by "Clangd"

**Comment by HighCommander4** [comment, marked as answer]
The reason is that clangd only loads the index when the compilation database is loaded, and the compilation database is only loaded when a source file is opened (because it is found based on the location of the source file).

There was some discussion [here](https://github.com/clangd/clangd/discussions/907#discussioncomment-1540740) of an option ("strong workspace mode") to have clangd treat the `rootUri` passed in the initialize request as the workspace root and load the compilation database there right after the initialize request. However, this has not yet been implemented.

**Reply by Yurii-huang** [reply]
Thank U.
I am now using libclang to parse the text of the source file. 
Now Clangd doesn't seem to support the operation, cause I convinced myself from the protocol of "didopen". 
Take the liberty to ask whether the Cland strong workspace supports this in the agreement ("strong workspace mode"). 
If not, it will be a long wait.

**Reply by Yurii-huang** [reply]
Text parsing that directly references libclang. It may be more in line with the different scenario requirements of the front end.

Considering that the IDE needs to integrate different languages, the "strong workspace mode" may also adopt the JSON-RPC communication mode.

However, backend programs may not directly use Clangd to add new functions
